nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/util/TextLineDemarcator.java

```diff
@@ -159,8 +159,11 @@ private int isEol(byte currentByte, int currentIndex) {
                 this.index = currentIndex + 1;
                 this.fill();
             }
-            currentByte = this.buffer[currentIndex + 1];
-            crlfLength = currentByte == '\n' ? 2 : 1;
+            crlfLength = 1;
+            if (this.bufferLength != -1) {
```

Comment:
After fill(), currentIndex can be equal to bufferLength - 1, so the next line can still cause an AIOOBE